sdk/python/kfp/components/_python_op.py

```diff
@@ -535,6 +538,15 @@ def _func_to_component_spec_v2(
 
         arguments.extend(arguments_for_input)
 
+    # Add output placeholders for return values from func.
+    func_outputs = [
+        output for output in component_outputs
+        if output._passing_style is None
+    ]
+    for output in func_outputs:
```

Comment:
JFYI: In the current lightweight components implementation, the paths for the by-value outputs are passed through the `----output-paths` argument which lists the paths.